lib/generators.nix

```diff
@@ -35,6 +35,7 @@ rec {
           ("generators.mkValueStringDefault: " +
            "${t} not supported: ${toPretty {} v}");
     in   if isInt      v then toString v
+    else if isFloat    v then toString v
```

Comment:
nope, not a good idea. Look at the bottom of the function, there is already a case for `isFloat`.

Reply:
Oh, didn't see that. Other than Nix not printing floats with many decimals, what's the problem with this? Having this case supported here will make things a lot easier, and I don't think there's a case where it can cause harm.

Reply:
Nix just silently removes decimals, so it’s really not an option to print them.

Reply:
Yeah but I don't think it's a problem. It seems that Nix prints 6 relevant digits, which should be enough for configurations, which is pretty much what Nix is for. I don't see how this could cause any harm.

Reply:
Silently removing *anything* is not an option. Big :-1: on this change.

Reply:
Well I need to show floats in this PR. I guess if you're against it here I'll just have to put it somewhere else where it can be used by the functions.

Reply:
https://github.com/NixOS/nixpkgs/pull/75584/commits/a39fe137ade65c9351740f2d9792674004f995fc

Reply:
As suggested by @emilazy on IRC, it's possible to detect when an inaccurate conversion would be done. Will do that and only then throw an error

Reply:
Did the thing in the latest commits now